src/components/molecules/math-operation/math-operation.component.tsx

```diff
@@ -60,41 +59,49 @@ const MathOperationComponent: React.FC<MathOperationProps> = ({ operationLabel,
         onOperationClick(inputs, setResult, setErrorMessage);
     };
 
-    return (
-        <div className="math-operation-container">
-            {/* Nawigacja */}
-            <div className="navigation">
-                <Link to="/math">
-                    <Button label="⬅" />
-                </Link>
-
-                {operation && <Label text={operationNames[operation]} />}
-            </div>
+return (
+    <Paper elevation={3} sx={{ p: 4, 
+    width: '60vw', 
+    display: 'flex', 
+    flexDirection: 'column', 
+    alignItems: 'center', 
+    backgroundColor: '#CFEDDB' }}>
+        {/* Nawigacja */}
+        <BaseButton component={Link} to="/math">
+            <ArrowBackIosIcon />
+        </BaseButton>
 
-            {/* Pola wejściowe */}
-            <div className="input-fields">
-                {inputs.map((input, index) => (
-                    <Input
-                        key={index}
+        {operation && 
```

Comment:
ten problem może tu nie istnieć, ale zostawiam Ci komentarz aby wskazać, że coś takiego może się stać przy takim zapisie:

https://javascript.plainenglish.io/how-to-stop-react-rendering-0-when-using-short-circuit-evaluation-3d20eb15dcde